src/main/kotlin/org/rust/ide/annotator/fixes/ExpandStructFieldsFix.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.utils.expand
+import org.rust.ide.utils.getTupleStructMissingFields
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.openapiext.buildAndRunTemplate
+import org.rust.openapiext.createSmartPointer
+import org.rust.stdext.mapToSet
+
+class ExpandStructFieldsFix(
+    element: PsiElement,
+    private val declaration: RsFieldsOwner
+) : LocalQuickFixAndIntentionActionOnPsiElement(element) {
+    override fun getText() = "Expand actual struct fields"
+
+    override fun getFamilyName() = text
+
+    override fun invoke(project: Project, file: PsiFile, editor: Editor?, startElement: PsiElement, endElement: PsiElement) {
+        if (startElement is RsPatStruct) {
+            val declarationFieldNames = declaration.fields.mapToSet { it.name }
+            val extraFields = startElement.patFieldList.filter { it.text !in declarationFieldNames }
```

Comment:
```suggestion
            val extraFields = startElement.patFieldList.filter { kind.fieldName!in declarationFieldNames }
```

If you use `element.text` (anywhere), it's probably a bug. This fails on `S { ref field }` or `S { field: f }`.